File: tsiolkovsky.m
function m0 = calculate_initial_mass(mf, Isp, delta_v)
    g0 = 9.81; % Acceleration due to gravity in m/s^2
    m0 = mf * exp(delta_v / (Isp * g0));
end

% Example values
mf = 1000; % final mass in kg
Isp = 300; % specific impulse in seconds
delta_v = 9500; % delta V for LEO in m/s

m0 = calculate_initial_mass(mf, Isp, delta_v);
fprintf('Initial mass (m0): %.2f kg\n', m0);

File: tsiolkovsky.cpp
#include <iostream>
#include <cmath>

double calculateInitialMass(double mf, double Isp, double delta_v) {
    const double g0 = 9.81; // Acceleration due to gravity in m/s^2
    double m0 = mf * exp(delta_v / (Isp * g0));
    return m0;
}

int main() {
    // Example values
    double mf = 1000.0; // final mass in kg
    double Isp = 300.0; // specific impulse in seconds
    double delta_v = 9500.0; // delta V for LEO in m/s

    double m0 = calculateInitialMass(mf, Isp, delta_v);
    std::cout << "Initial mass (m0): " << m0 << " kg" << std::endl;

    return 0;
}
